crates/pallet-domains/src/lib.rs

```diff
@@ -1855,6 +1855,8 @@ impl<T: Config> Pallet<T> {
     }
 
     /// Returns the block number of oldest execution receipt.
+    // FIXME: the `oldest_receipt_number` may not be correct if fraud proof is submitted
```

Comment:
maybe the way we derive this is not stable. We should instead track the confirmed domain block number and then return the immediate uncornfirmed block's ER as oldest receipt number instead of using head_receipt number since, like you mentioned, is prone to FraudProof pruning

Reply:
Yeah, we should explicitly track either the confirmed domain block number or the oldest receipt number (which does not make much difference IMO) instead of deriving from the head receipt number.